src/main/java/org/opensearch/knn/search/NativeEngineKnnFloatVectorQuery.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.search;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.QueryTimeout;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.KnnFloatVectorQuery;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.TotalHits;
+import org.apache.lucene.search.knn.KnnCollectorManager;
+import org.apache.lucene.util.BitSet;
+import org.apache.lucene.util.BitSetIterator;
+import org.apache.lucene.util.Bits;
+import org.opensearch.knn.index.query.KNNQuery;
+import org.opensearch.knn.index.query.KNNWeight;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class NativeEngineKnnFloatVectorQuery extends KnnFloatVectorQuery {
```

Comment:
if I am not wrong, KnnFloatVectorQuery just checks the FiltersIds<=K to do the exact search but not the other condition like AdvancedFilterThreshold. Thats a bumer.

Also, currently the way ParentIdsFilters and radial search work for Lucene VectorQuery and that of NativeEngine query is different.


Another thing, by extending KnnFloatVectorQuery, we have just reused the DocAndScore query if I am not wrong. Anything else that we reused here which I am missing?

Reply:
> if I am not wrong, KnnFloatVectorQuery just checks the FiltersIds<=K to do the exact search but not the other condition like AdvancedFilterThreshold. Thats a bumer.

Correct, losing support for a setting. so now backward compatible

> Also, currently the way ParentIdsFilters and radial search work for Lucene VectorQuery and that of NativeEngine query is different.

- parentIds: I missed this one, nice catch
- radial: this needs some thought so didn't plug it in here

> Anything else that we reused here which I am missing?

Yeah DocAndScore and some boiler plate filtering